sql/src/main/java/io/crate/analyze/relations/RelationSplitter.java

```diff
@@ -203,17 +202,75 @@ private void processWhere() {
                 qs.where(qs.where().add(queryForRelation));
             }
         }
+
+        convertImplicitJoinConditionsToJoinPairs(joinPairs, splitQueries);
         if (splitQueries.isEmpty()) {
             querySpec.where(WhereClause.MATCH_ALL);
         } else {
             Symbol newQuery = AndOperator.join(splitQueries.values());
             JoinConditionValidator.validate(newQuery);
             querySpec.where(new WhereClause(newQuery));
         }
+
+        processJoinPairs();
+    }
+
+    private void processJoinPairs() {
+        for (JoinPair joinPair : joinPairs) {
+            if (joinPair.condition() != null) {
+                JoinConditionValidator.validate(joinPair.condition());
+                joinConditions.add(joinPair.condition());
+                relationPartOfJoinConditions.add(joinPair.left());
+                relationPartOfJoinConditions.add(joinPair.right());
+            }
+        }
+    }
+
+    @VisibleForTesting
+    static void convertImplicitJoinConditionsToJoinPairs(List<JoinPair> joinPairs,
```

Comment:
as this is only sometimes possible - shall we rename to "maybeConvertImplicitConditionsToJoinPairs"?

Reply:
hm... I don't like "maybe", maybe "tryTo" is better?